be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -166,12 +168,20 @@ Status BetaRowsetWriter::flush_src_rssids(uint32_t segment_id) {
 }
 
 HorizontalBetaRowsetWriter::HorizontalBetaRowsetWriter(const RowsetWriterContext& context)
-        : BetaRowsetWriter(context), _segment_writer(nullptr) {}
+        : BetaRowsetWriter(context), _segment_writer(nullptr), _final_merge_start_write(false) {}
 
 HorizontalBetaRowsetWriter::~HorizontalBetaRowsetWriter() {
     // TODO(lingbin): Should wrapper exception logic, no need to know file ops directly.
-    if (!_already_built) {       // abnormal exit, remove all files generated
-        _segment_writer.reset(); // ensure all files are closed
+    if (!_already_built) { // abnormal exit, remove all files generated
+        if (_segment_writer != nullptr) {
+            // close the latest write file
+            _segment_writer->close();
+            _segment_writer.reset();
+        }
+        // sync the dir
+        if (_num_rows_written > 0 && !_already_sync_dir) {
+            _fs->sync_dir(_context.rowset_path_prefix);
```

Comment:
why sync_dir?

Reply:
To ensure the previously written segment files and directories have been flushed to the disk, after that we can make sure to find the file we want to delete, may be unnecessary?

Reply:
@chaoyli do you know why ?

Reply:
dir is also one file, if not sync, the file may be not exist in dir?